### DISCUSSION TITLE
Wrapper for ObjectPageSection

### DISCUSSION THREAD

**Discussion by reqyou** [UI5/webcomponents-react#3222] [Ideas]
Hello,
It would be nice to have a wrapper component for the ObjectPageSection to be able to have a slim ObjectPage and to be able to define all its content outside in separate components. 

Best

**Comment by Lukas742** [comment]
Hi @reqyou 

it's recommended that only the children of each section are rendered in a separate component. Still, it's possible to do the same with a section but you have to make sure, that all props are passed to the wrapper component and then are applied to the section inside of it.

I created a [codeSandbox](https://codesandbox.io/s/objectpage-story-forked-e1c9y7?file=/src/App.js) that should explain this in more detail.

**Reply by reqyou** [reply]
Hey @Lukas742,
thanks for the prompt answer. Can you please go more in detail, why this is recommended? 🙇‍♀️ 

Thanks

**Reply by Lukas742** [reply]
The `ObjectPage` performs some calculations based on its children and that is why we do not currently openly document a way to use components other than the `ObjectPageSection` and `ObjectPageSectionSubSection`. The reason behind this is that we cannot offer support for custom implementations like this, even though they will (if correctly implemented) most likely work. 

If we find the time we will look into improving this, e.g. with a context based approach, but currently this is not on our roadmap.
